manifests/dropin_file.pp

```diff
@@ -69,11 +71,26 @@
     show_diff               => $show_diff,
   }
 
+  if $daemon_reload {
+    ensure_resource('systemd::daemon_reload', $name)
```

Comment:
I think this should be one per unit, in case there are multiple dropin_files per unit. Possibly also normalized to a short name (like on line 88)
```suggestion
    ensure_resource('systemd::daemon_reload', $unit)
```

Reply:
That works. Fewer resources in the catalog that way.